common/utils/src/main/java/org/apache/spark/QueryContext.java

```diff
@@ -45,4 +48,13 @@ public interface QueryContext {
 
     // The corresponding fragment of the query which throws the exception.
     String fragment();
+
+    // The Spark code (API) that caused throwing the exception.
+    String code();
```

Comment:
shall we reuse the `fragment` function to return the code fragment?

Reply:
done